### ISSUE TITLE
How to utilize "data" when the app is launched from a notification

### ISSUE THREAD

**Issue by davecoffin** [firebase/quickstart-ios#173]
So, my backend is sending push notifications, for example like this:

```
$data = array(
    "notification" =>  array(
        "title" => $title,
        "body" => $message,
        "sound" => true,
        "badge" => "1"
    ),
    "to" => $tok,
    "content_available"=>true
);
$data['data'] = array(
    "type"=>"foo"
);
```

If I receive a notification when the app is open, I can inspect the contents of data and do something, like offer to bring the user to foo page. But, if the user receives the typical iOS notification because the app is in the background, when the user opens the app by tapping the notification, it just opens the app. Theres no way I can see to inspect data to bring them to the appropriate page based on the data. Is this possible?

**Comment by ianbarber** [comment]
It should be turning up the in UNNotificationResponse object in the didReceiveNotificationResponse callback.

**Comment by jonnyleeharris** [comment]
I'm struggling with this too.

No call backs are fired when the app is brought to the foreground from a notification.

The correct callbacks are fired when the same notification is sent when the app is already in the foreground.

**Comment by jonnyleeharris** [comment]
I just managed to get it working by implementing this method:

`@available(iOS 10.0, *)
	func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
		log.debug("userNotificationCenter didReceive")
		let userInfo = response.notification.request.content.userInfo
		self.notificationHandler?.handleNotification(payload: userInfo)
	}`

**Comment by kroikie** [comment]
Closing this issue since it is solved.
